### PR TITLE
FUSETOOLS2-2232: UI tests refactoring

### PR DESCRIPTION
**Pull request by unsortedhashsets**
- Refactoring of runner and suite
- Added simple files (xml, java, yaml, yml)
- Updated install test based on https://github.com/camel-tooling/camel-dap-client-vscode/pull/516
- Works tests for Basic (with webView logs check) and Dev (with update/reload, terminal logs check) modes for java, xml, yaml, yml
- Tests for XML are skipped due - [FUSETOOLS2-2238](https://issues.redhat.com/browse/FUSETOOLS2-2238)

[FUSETOOLS2-2232](https://issues.redhat.com/browse/FUSETOOLS2-2232)

### CODE REVIEW

**Review by apupier — APPROVED** [review]

**Comment by apupier** [comment]
the blocked job `Code scanning results / API` is strange , it has a different name than on other PR (`Code scanning results / SonarCloud`) I think we can ignore it

File: src/ui-test/tests/dev_mode.test.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			if (extension == 'java') {
+				await driver.wait(() => { return updateFileText(consts.integrationFileName, `${consts.integrationFileName}Copy`); });
+			}
+		});
+
+		after(async function () {
+			this.timeout(consts.TIMEOUT_60_SECONDS);
+			await new EditorView().closeAllEditors();
+			await sidebarIntegrationRemove(driver, consts.extensionName, `${consts.integrationFileName}-copy`);
+			await driver.wait(() => { return cleanOutputView(); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:94]: missing default timeout value

File: src/ui-test/tests/dev_mode.test.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+					return false;
+				}
+			}, 60000);
+		});
+
+		it(`Select ${consts.startIntegration} in the popup menu`, async function () {
+			this.timeout(consts.TIMEOUT_15_SECONDS);
+			const section = await new SideBarView().getContent().getSection('resources');
+			const item = await section.findItem(`${consts.integrationFileName}Copy.${extension}`) as DefaultTreeItem;
+			await driver.wait(() => { return contextMenuItemClick(item, consts.startIntegration); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:110]: missing default timeout value

File: src/ui-test/tests/dev_mode.test.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		});
+
+		it(`Initial integration pod started`, async function () {
+			this.timeout(consts.TIMEOUT_30_SECONDS);
+			await driver.wait(() => { return outputViewHasText(consts.initialPodReadyMessage); });
+		});
+
+		it(`Initial integration output contains - ${initialMessage}`, async function () {
+			this.timeout(consts.TIMEOUT_30_SECONDS);
+			await driver.wait(() => { return outputViewHasText(initialMessage); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:125]: missing default timeout value

File: src/ui-test/tests/dev_mode.test.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		it(`Integration output does not contain ASCI`, async function () {
+			this.timeout(consts.TIMEOUT_5_SECONDS);
+			this.id = 'independent'
+			assert.isTrue(await textDoesNotContainAsci('OutputView'));
+		});
+
+		it(`Integration exists in ${consts.extensionName} sidebar`, async function () {
+			this.timeout(consts.TIMEOUT_15_SECONDS);
+			const content = new SideBarView().getContent();
+			await driver.wait(() => { return viewHasItem(content, consts.extensionName, `${consts.integrationFileName}-copy`.toLowerCase()); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:137]: missing default timeout value

File: src/ui-test/tests/dev_mode.test.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+		});
+
+		it(`Update ${consts.integrationFileName}Copy.${extension} with new message`, async function () {
+			this.timeout(consts.TIMEOUT_30_SECONDS);
+			await driver.wait(() => { return updateFileText(initialMessage, updatedMessage); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:142]: missing default timeout value

File: src/ui-test/tests/dev_mode.test.ts [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+		});
+
+		it(`Updated integration pod started`, async function () {
+			this.timeout(consts.TIMEOUT_30_SECONDS);
+			await driver.wait(() => { return outputViewHasText(consts.updatedPodReadyMessage); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:147]: missing default timeout value

File: src/ui-test/tests/dev_mode.test.ts [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+		});
+
+		it(`Updated integration output contains - ${updatedMessage}`, async function () {
+			this.timeout(consts.TIMEOUT_30_SECONDS);
+			await driver.wait(() => { return outputViewHasText(updatedMessage); });
```
Comment (djelinek) [inline src/ui-test/tests/dev_mode.test.ts:152]: missing default timeout value

**Review by djelinek — COMMENTED** [review]
you can also uninstall vscode-extension-tester devDependency

**Pull request merged**